test/e2e/gitops_service_test.go

```diff
@@ -587,7 +587,7 @@ func validateGrantingPermissionsByLabel(t *testing.T) {
 	err = cmd.Run()
 	assertNoError(t, err)
 
-	err = wait.Poll(time.Second*1, time.Second*180, func() (bool, error) {
+	err = wait.Poll(time.Second*1, time.Minute*5, func() (bool, error) {
```

Comment:
I dont think increase the timeout to 5 minutes is a right thing to do in this case. We should identify the actual point of intermittent or flaky behavior :)

Reply:
This is a wait poll timeout which means the longest it would wait.  Do we see any evidence that it can normally take more than 100 seconds?

Reply:
Sometimes it takes around 6-7 minutes to reach a healthy state. Not sure what's causing this delay. I see the timeout is now increased to 10 mins by PR #161

Reply:
Agreed that increasing it isn't the best solution: personally I investigated the issue and ran out of leads (for example was never able to reproduce on clusterbot cluster with e2e tests run in a loop), and increased it more out of frustration then good sense :upside_down_face:.

Reply:
I couldn't reproduce it manually as well. The app reached a healthy state seamlessly without any delay

Reply:
/lgtm
/approve